zenoh/src/net/routing/hat/linkstate_peer/network.rs

```diff
@@ -752,7 +753,7 @@ impl Network {
                 tracing::trace!("Update edge (link) {} {}", self.graph[self.idx].zid, zid);
                 self.update_edge(self.idx, idx);
             }
-            self.graph[self.idx].links.push(zid);
+            self.graph[self.idx].links.insert(zid);
```

Comment:
Here there is a semantic change where before the same `ZenohId` could be present multiple times in the `Vec<ZenohIdProto>` while now it can be present only once.
Maybe is it worth adding a debug log if `insert(zid)` returns `Some`? It would allow to detect if something else is going wrong. What do you think?